File: meldingsvarsler-fasit-forslag/src/main/java/no/sonat/meldingsvarsler/facebook/FacebookProsessor.java
package no.sonat.meldingsvarsler.facebook;

import no.sonat.meldingsvarsler.abonnent.Abonnent;
import no.sonat.meldingsvarsler.Melding;
import no.sonat.meldingsvarsler.MeldingProsessor;

public class FacebookProsessor implements MeldingProsessor {
    @Override
    public boolean haandtererUtsendingAv(Melding melding) {
        return melding instanceof FacebookMelding;
    }

    @Override
    public void sendMelding(Abonnent abonnent, Melding melding) {
        sendFaceMelding(((FacebookAbonnent) abonnent), (FacebookMelding) melding);
    }

    public void sendFaceMelding(FacebookAbonnent facebookAbonnent, FacebookMelding facebookMelding) {
        System.out.println("Sender Facebook melding til "
                + facebookAbonnent.navn()
                + " => " + facebookMelding.melding());
    }
}

File: meldingsvarsler-fasit-forslag/App/FacebookProsessor.cs
using System;

namespace App {
    public class FacebookProsessor : IMeldingProsessor {
    
        public bool HaandtererUtsendingAv(IMelding melding) {
            return melding is FacebookMelding;
        }

    
        public void SendMelding(IAbonnent abonnent, IMelding melding) {
            SendFaceMelding(((FacebookAbonnent) abonnent), (FacebookMelding) melding);
        }

        public void SendFaceMelding(FacebookAbonnent facebookAbonnent, FacebookMelding facebookMelding) {
            Console.Out.WriteLine("Sender Facebook melding til "
                                + facebookAbonnent.Navn
                                + " => " + facebookMelding.Melding);
        }
    }
}
